pkg/limayaml/validate.go

```diff
@@ -372,6 +372,45 @@ func Validate(y *LimaYAML, warn bool) error {
 		}
 	}
 
+	if y.Rosetta.Enabled != nil {
+		if *y.Rosetta.Enabled && *y.VMType != VZ {
+			return fmt.Errorf("field `rosetta.enabled` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.NestedVirtualization != nil && *y.NestedVirtualization {
+		if *y.VMType != VZ {
+			return fmt.Errorf("field `nestedVirtualization` can only be enabled for VMType %q; got %q", VZ, *y.VMType)
+		}
+	}
+
+	if y.MountType != nil {
+		if err := validateMountType(y); err != nil {
+			return err
+		}
+	}
```

Comment:
`y.MountType` cannot be `nil` after `FillDefaults()` has been called. Same way you didn't need to check if `y.VMType` is `nil`.

```suggestion
	if err := validateMountType(y); err != nil {
		return err
	}
```